### DISCUSSION TITLE
[C#] How to set a default Auth Type

### DISCUSSION THREAD

**Discussion by maximeozenne** [scalar/scalar#3252] [Q&A] [answered]
Hello,

Using Scalar in dotnet, I'm trying to set the `Bearer` auth type as default. I only use this type of authentication, so I was wondering if it can be pre-selected ?

![CleanShot 2024-09-25 at 19 20 10](https://github.com/user-attachments/assets/40ca8196-ec12-4e73-a24e-4aeb684ea802)

Thanks for your help!

**Comment by xC0dex** [comment]
You can set it by using the following method:
```c#
options.WithPreferredScheme("Bearer");
```

In this case, `Bearer` is the name of the defined security scheme.

**Reply by maximeozenne** [reply]
Even with this option, the Auth Type is still on `None` by default :/

**Reply by xC0dex** [reply]
Strange. I'll check that out later. It was working as expected in the `Playground` project yesterday.

**Reply by maximeozenne** [reply]
Hello @xC0dex !

I've pushed my test project to GitHub, hope it can helps you : https://github.com/maximeozenne/TestSwaggerNet9

In the `Program.cs`, line 106, I added the `.WithPreferredScheme("Bearer");` :

https://github.com/maximeozenne/TestSwaggerNet9/blob/b5cc1a847bd494ab995dcea9c2938de3491341bb/TestSwaggerNet9/Program.cs#L106

But when you run the app and go to any endpoint, (I use the last endpoint of the starting page, called "Say Hello to someone"), the Authentication Type is still on `None` :

![CleanShot 2024-09-26 at 16 40 58](https://github.com/user-attachments/assets/6c9caa88-38e6-48ab-9a2f-8e779baafc9f)

Thank you very much for your active help on all my requests!!

**Reply by xC0dex** [reply]
Thanks! I just saw that the same issue was reported in Discord. It looks like the `preferredSecurityScheme` property doesn't work as expected.

**Reply by xC0dex** [reply]
Here it is teh related issue https://github.com/scalar/scalar/issues/3264

**Reply by maximeozenne** [reply]
Thanks, I subscribed and I'll wait for it to be resolved. Again, thanks for your work!

**Reply by maximeozenne** [reply]
Hi again @xC0dex!

The issue you mentioned earlier is marked as completed, but even using the new `1.2.5` release, the `preferredSecurityScheme` does not seem to be fixed.

I've updated my repo if you want to take a peek at my code, maybe I'm doing something wrong? => https://github.com/maximeozenne/TestSwaggerNet9

**Reply by xC0dex** [reply]
I agree, the issue also still exist on my end.

**Reply by xC0dex** [reply, marked as answer]
I create an issue https://github.com/scalar/scalar/issues/3295. I would rather not rule out that I implemented something wrong in the `Scalar.AspNetCore` package.

**Reply by maximeozenne** [reply]
Thank you very much @xC0dex, you're a lifesaver!

**Reply by xC0dex** [reply]
Hey @maximeozenne, the issue should be fixed! 🚀

**Reply by maximeozenne** [reply]
Hello @xC0dex ! It works !! Thank you very much !
